astroplan/core.py

```diff
@@ -22,109 +38,152 @@
 
 class Observer(object):
     """
-    Some comments.
+    A container for information about an observatory's location and environment.
     """
 
-    def __init__(self, name, longitude, latitude, elevation, pressure,
-                 relative_humidity, temperature, timezone, description=None):
+    def __init__(self, name=None, location=None, latitude=None, longitude=None,
+                 elevation=None, timezone='UTC', pressure=None,
+                 relative_humidity=None, temperature=None, description=None):
         """
-        Initializes an Observer object.
-
-        <longer description>
-
         Parameters
         ----------
         name : str
             A short name for the telescope, observatory or location.
 
-        longitude : str or `~astropy.units.quantity`?
-            The longitude of the observing location.
+        location : `~astropy.coordinates.EarthLocation`
+            The location (latitude, longitude, elevation) of the observatory.
+
+        longitude : float, str, `~astropy.units.Quantity` (optional)
+            The longitude of the observing location. Should be valid input for
+            initializing a `~astropy.coordinates.Longitude` object.
 
-        latitude : str or `~astropy.units.quantity`?
-            The latitude of the observing location.
+        latitude : float, str, `~astropy.units.Quantity` (optional)
+            The latitude of the observing location. Should be valid input for
+            initializing a `~astropy.coordinates.Latitude` object.
 
-        elevation : `~astropy.units.Quantity`
+        elevation : `~astropy.units.Quantity` (optional)
             The elevation of the observing location, with respect to sea
-            level.
+            level. Defaults to zero meters.
 
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
+        pressure : `~astropy.units.Quantity` (optional)
+            The ambient pressure. Defaults to zero (i.e. no atmosphere).
 
-        relative_humidity : float
+        relative_humidity : float (optional)
             The ambient relative humidity.
 
-        temperature : `~astropy.units.Quantity`
+        temperature : `~astropy.units.Quantity` (optional)
             The ambient temperature.
 
-        timezone : WHAT TYPE IS pytz.timezone ?
-            The local timezone.
+        timezone : str or `datetime.tzinfo` (optional)
+            The local timezone, as either an instance of `pytz.timezone()` or
```

Comment:
The way you've phrased it here you're duplicating information because you're just saying the type.  Maybe instead "The local timezone to assume.  If a string, it will be passed through \`pytz.timezone()\` to produce the `timezone` object.